cucumber-messages/go/messages.proto

```diff
@@ -4,19 +4,57 @@ package io.cucumber.messages;
 option ruby_package = "Cucumber.Messages";
 option go_package = "messages";
 
-// All messages sent between processes must be of type Wrapper
-message Wrapper {
+// When removing a field, comment it out, rather than deleting the line.
+// When adding a field, increment the number by one.
+
+// All messages sent between gherkin or formatters must be of type EventWrapper
+message EventWrapper {
   oneof message {
+    // Gherkin messages
     Source source = 1;
     GherkinDocument gherkinDocument = 2;
     Pickle pickle = 3;
     Attachment attachment = 4;
+    // Execution Messages
     TestCaseStarted testCaseStarted = 5;
     TestStepStarted testStepStarted = 6;
     TestStepFinished testStepFinished = 7;
     TestCaseFinished testCaseFinished = 8;
     TestHookStarted testHookStarted = 9;
     TestHookFinished testHookFinished = 10;
+    PickleAccepted pickleAccepted = 11;
+    PickleRejected pickleRejected = 12;
+    TestCasePrepared testCasePrepared = 13;
+    TestRunStarted testRunStarted = 14;
+    TestRunFinished testRunFinished = 15;
+  }
+  google.protobuf.Timestamp timestamp = 16;
+}
+
+// All messages sent between cucumber-engine and its calling process must be of type CommandWrapper
+message CommandWrapper {
+  string id = 1;
```

Comment:
This could be removed and instead we add a `actionID` to each command under the comment: `// Sent by cucumber-engine to perform actions`